packages/ec-core/src/operator/selector/lexicase.rs

```diff
@@ -95,3 +100,181 @@ where
             .context("The pool of candidates was empty")
     }
 }
+
+#[cfg(test)]
+#[rustversion::attr(before(1.81), allow(clippy::unwrap_used))]
+#[rustversion::attr(
+    since(1.81),
+    expect(
+        clippy::unwrap_used,
+        reason = "Panicking is the best way to deal with errors in unit tests"
+    )
+)]
+mod tests {
+    use prop::collection;
+    use proptest::prelude::*;
+
+    use super::*;
+    use crate::individual::ec::EcIndividual;
+
+    #[test]
+    fn single_best_single_error() {
+        // 9 is the sole winner
+        let population = vec![5, 8, 9, 6, 3, 2, 0]
+            .into_iter()
+            .enumerate()
+            .map(|(id, e)| EcIndividual::<usize, TestResults<i32>>::new(id, TestResults::from([e])))
```

Comment:
Same as below here as well
```suggestion
            .map(|v| [v])
            .map(TestResults::<i32>::from)
            .enumerate()
            .map(EcIndividual::new)
```